torch/_library/define.py

```diff
@@ -0,0 +1,332 @@
+import inspect
+import sys
+from typing import Callable, Dict, List, Optional, Sequence, Union
+
+from .. import _C, _library, library, Tensor
+
+
+device_types_t = Optional[Union[str, Sequence[str]]]
+
+
+def def_blackbox(*, mutated_args: Sequence[str], types: device_types_t = None):
+    """Wraps a function into a custom operator.
+
+    Treats the function as a black-box (that is, PyTorch will never peek into
+    the function). The function must have type hints; these are needed
```

Comment:
This description doesn't seem clear. In particular, suppose I am a clueless user. Why would I not want PyTorch to peek into the function?